### PR TITLE
-Added difficulty level buttons nested within 1P mode button.

### LINKED ISSUE #2: Easy, Medium and Hard AI
Make the AI difficulty modes. The original ticket for implementing the medium AI was closed prematurely, so that issue has been added to this one

This should be implemented using a queue. This should be a button option on new game that is like "one player" or something and a drop down for AI difficulty.

For the easy AI, the queue should be empty, and should always draw randomly where to hit.

For the medium AI, use the same logic to randomly shoot as the easy AI. On a hit, it should iterate through the 4 orthogonal squares around it until it gets another hit. upon the second hit, it keeps hitting in that direction until the ship is sunk. If it gets a miss along that path and the ship isn't sunk, it should shoot on the opposite side of the 1st hit and hit in that direction until the ship is sunk. This will be the most difficult mode to implement.

For the hard AI, populate the queue with your ship positions and pop them as you go.

Finished when all three are implemented.

### PR DESCRIPTION
**Pull request by Platnumrcn14**
**homeScreen.html**
-Implemented a difficulty selector option nested in the 1P mode button of the main menu

 

**server.js**
-created the framework for homeScreen.html to communicate the difficulty option selected to the server.js

**battleshipRound.js**
-Implemented Hard mode for AI that will iterate through the game board for a hittable grid position. Once found will hit that position. 
-Medium mode nested within easy mode

### CODE REVIEW

File: back/server/server.js [hunk 1]
```diff
@@ -91,9 +91,17 @@ io.on('connection', (socket) => {
     socket.emit('getClientId', {});
 
     // Placeholder for future AI implementation
-    socket.on('initiateSinglePlayer', (numShips) => {
+    socket.on('initiateSinglePlayer', (numShips, difficulty) => {
         // TODO: ALLOW SPECIFYING DIFFICULTY
```
Comment (jacobwilki01) [inline back/server/server.js]: remove todo comment

File: front/homeScreen/homeScreen.html
Comment (jacobwilki01) [inline front/homeScreen/homeScreen.html:1]: why'd you change the spacing of this file so much?

File: back/server/modules/battleshipRound.js [hunk 2]
```diff
@@ -180,6 +186,17 @@ class BattleshipRound {
         return {x:x, y:y}
     }
 
+    targetNextShipTile(opMap) {
```
Comment (jacobwilki01) [inline back/server/modules/battleshipRound.js]: this is an inefficient way of doing this. Your current implementation has us iterate through the grid **every single time** its the AI's turn, and checks tiles we know are hit already.

Instead, lets generate a list of "tiles to hit" at the start of the first AI turn that is stored in this class. Then, whenever its the AI's turn, grab the next item in the list, remove it from the list, and return that tile.

**Review by jacobwilki01 — CHANGES_REQUESTED** [review]

**Review by eBerkley — APPROVED** [review]
Looks good to me.

File: back/server/modules/battleshipRound.js [hunk 3]
```diff
@@ -180,6 +187,27 @@ class BattleshipRound {
         return {x:x, y:y}
     }
 
+    targetShipTiles(opMap) {
+        // If the target list is empty, we need to scan the grid and populate it
```
Comment (jacobwilki01) [inline back/server/modules/battleshipRound.js:363]: Let's invert the order of these. So:

if (this.aiTargetList.Length > 0) {
  return this.aiTargetList.shift();
}

THEN generate the list and return the first value without the if statement. That removes the need for the length check of 0 and some of the nesting.

**Review by jacobwilki01 — APPROVED** [review]

**Review by richardmoser — APPROVED** [review]

**Review by modmo22 — APPROVED** [review]
Looks great!

**Pull request merged**